.github/workflows/release-shell-pkg.yaml

```diff
@@ -0,0 +1,46 @@
+name: Publish Shell Package
+
+on:
+  push:
+    tags:
+        - 'shell-pkg-v*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        persist-credentials: false
+
+    - uses: actions/setup-node@v3
+      with:
+        node-version: '16.x'
+
+    - name: Validate Plugin build system
+      run: ./shell/scripts/test-plugins-build.sh
+      shell: bash
+
+    - name: Reset repository (file system)
+      run: | 
+        git reset --hard HEAD
+        echo $YARN_REGISTRY
+        cat ./shell/package.json
```

Comment:
I'm a little confused by this step, why is is required to run `git reset --hard HEAD` before we publish?

Reply:
Because the step `Validate Plugin build system` messes with the file system in the pipeline (especially the version number on `package.json` so that it can be published to verdaccio to do it's checks).

We don't want to mess with that, so resetting the repo locally seems to be a good candidate.

Another alternative would be to do another checkout step (I think... I don't believe I tested it).

This is a keeper

Reply:
Shall we leave a comment that describes this behavior for posterity? As it stands, this is the only thing in this action that is potentially confusing.

Reply:
Added :)